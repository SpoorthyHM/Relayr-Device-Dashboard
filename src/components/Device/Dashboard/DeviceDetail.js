import React from 'react';
import Switch from "react-switch";

const deviceDetail = (props) => {
    let row ='';
    if(props.device) {
        row = <tr>
                <td>{props.device.name}</td>
                <td>{props.device.unit || 'NA'}</td>
                <td>{props.device.value}</td>
                <td>{props.device.timestamp}</td>
                <td>
                    <Switch onColor='#144c52' offColor='#ccc'
                        id={props.device.name}
                        className="status-switch"
                        onChange={(checked, event, id) => props.onStatusChange(checked, id)}
                        checked={props.device.active || false} />
                </td>
            </tr>
    }
    return row;
}

export default deviceDetail;