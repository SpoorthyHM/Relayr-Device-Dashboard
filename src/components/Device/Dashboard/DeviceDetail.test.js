import React from 'react';
import DeviceDetail from './DeviceDetail';
import { shallow } from 'enzyme';

describe('DeviceDetails', () => {
    let data = {
        "device": 
            {
                "name": "acceleration_x",
                "unit": "m/s2",
                "value": 25.993848858558,
                "timestamp": 1556550441396,
                "active": true
            }
        }

    let noData = {"device" : {}};
    it('should render correctly with data', () => {
        const component = shallow(<DeviceDetail {...data}/>);
        component.setProps(data);
        expect(component).toMatchSnapshot();
      });

    it('should render correctly with empty props', () => {
        const component = shallow(<DeviceDetail {...noData}/>);
        component.setProps(noData);
        expect(component).toMatchSnapshot();
    });
});
