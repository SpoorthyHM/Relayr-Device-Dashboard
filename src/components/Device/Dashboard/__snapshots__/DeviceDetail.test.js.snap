// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceDetails should render correctly with data 1`] = `ShallowWrapper {}`;

exports[`DeviceDetails should render correctly with empty props 1`] = `ShallowWrapper {}`;
